sql/src/main/java/io/crate/analyze/GeneratedColumnComparisonReplacer.java

```diff
@@ -148,7 +148,7 @@ private Symbol wrapInGenerationExpression(Symbol wrapMeLikeItsHot, Reference gen
         private Map<ReferenceInfo, GeneratedReferenceInfo> extractGeneratedReferences(DocTableInfo tableInfo) {
             ImmutableMap.Builder<ReferenceInfo, GeneratedReferenceInfo> builder = ImmutableMap.builder();
             for (GeneratedReferenceInfo referenceInfo : tableInfo.generatedColumns()) {
-                if (referenceInfo.referencedReferenceInfos().size() == 1) {
+                if (tableInfo.partitionedByColumns().contains(referenceInfo) && referenceInfo.referencedReferenceInfos().size() == 1) {
```

Comment:
.size() check is probably cheaper than the contains check.. so it would probably make sense to reverse the order of the checks?